Cubical/Homotopy/Connected.agda

```diff
@@ -132,63 +143,61 @@ isOfHLevelPrecomposeConnected : ∀ {ℓ ℓ' ℓ''} (k : HLevel) (n : HLevel)
 isOfHLevelPrecomposeConnected zero n P f fConn =
   elim.isEquivPrecompose f n P fConn .equiv-proof
 isOfHLevelPrecomposeConnected (suc k) n P f fConn t =
-    isOfHLevelPath'⁻ k
-      (λ {(s₀ , p₀) (s₁ , p₁) →
-        subst (isOfHLevel k) (sym (fiber≡ (s₀ , p₀) (s₁ , p₁)))
-          (isOfHLevelRetract k
+  isOfHLevelPath'⁻ k
+    λ {(s₀ , p₀) (s₁ , p₁) →
+      isOfHLevelRetract k
+         (Iso.inv ΣPathIsoPathΣ)
+         (Iso.fun ΣPathIsoPathΣ)
+         (Iso.rightInv ΣPathIsoPathΣ)
+         (subst (isOfHLevel k)
+           (sym (fiberPath (s₀ , p₀) (s₁ , p₁)))
+           (isOfHLevelRetract k
             (λ {(q , α) → (funExt⁻ q) , (cong funExt⁻ α)})
             (λ {(h , β) → (funExt h) , (cong funExt β)})
             (λ _ → refl)
             (isOfHLevelPrecomposeConnected k n
               (λ b → (s₀ b ≡ s₁ b) , isOfHLevelPath' (k + n) (P b .snd) _ _)
               f fConn
-              (funExt⁻ (p₀ ∙∙ refl ∙∙ sym p₁))))})
+              (funExt⁻ (p₀ ∙∙ refl ∙∙ sym p₁)))))}
 
 indMapEquiv→conType : ∀ {ℓ} {A : Type ℓ} (n : HLevel)
                    → ((B : TypeOfHLevel ℓ n)
                       → isEquiv (λ (b : (fst B)) → λ (a : A) → b))
                    → isConnected n A
-indMapEquiv→conType {A = A} n BEq =
-  transport (λ i → isContr (hLevelTrunc n (typeToFiber A (~ i))))
-            (elim.isConnectedPrecompose (λ _ → tt) n
-                                        (λ P → isEquiv→hasSection ((compEquiv ((λ Q → Q tt) , isoToIsEquiv (helper P))
-                                                                              (_ , BEq (P tt)) .snd )))
-                                        tt)
-  where
-  helper : ∀ {ℓ'} (P : Unit → TypeOfHLevel ℓ' n)
-         → Iso ((b : Unit) → fst (P b)) (fst (P tt))
-  Iso.fun (helper P) = λ Q → Q tt
-  Iso.inv (helper P) = λ a _ → a
-  Iso.rightInv (helper P) b = refl
-  Iso.leftInv (helper P) b = refl
+indMapEquiv→conType {A = A} zero BEq = isContrUnit*
+indMapEquiv→conType {A = A} (suc n) BEq =
+  isContrRetract
+    (map {n = (suc n)} (Iso.fun (typeToFiberIso A)))
+    (map (Iso.inv (typeToFiberIso A)))
+    (Trunc.elim (λ _ → isOfHLevelPath (suc n) (isOfHLevelTrunc (suc n)) _ _) (λ _ → refl))
+    (elim.isConnectedPrecompose (λ _ → tt) (suc n)
+                                (λ P → ((λ a _ → a) ∘ invIsEq (BEq (P tt)))
+                               , λ a → equiv-proof (BEq (P tt)) a .fst .snd)
+                                tt)
 
 isConnectedPath : ∀ {ℓ} (n : HLevel) {A : Type ℓ}
   → isConnected (suc n) A
   → (a₀ a₁ : A) → isConnected n (a₀ ≡ a₁)
-isConnectedPath n connA a₀ a₁ =
-  subst isContr (PathIdTrunc _)
+isConnectedPath zero connA a₀ a₁ = isContrUnit*
+isConnectedPath (suc n) connA a₀ a₁ =
+  isContrRetract
```

Comment:
It seems like this proof pattern is appearing a lot. You apply "isContrRetract" (or similar lemmas) so some iso which you need to decompose. Why not prove this once and for all as a lemma "isContrRetractFromIso" or something?